library/core/src/task/wake.rs

```diff
@@ -195,16 +236,33 @@ impl<'a> Context<'a> {
     #[must_use]
     #[inline]
     pub const fn from_waker(waker: &'a Waker) -> Self {
-        Context { waker, _marker: PhantomData, _marker2: PhantomData }
+        ContextBuilder::from_waker(waker).build()
     }
 
     /// Returns a reference to the [`Waker`] for the current task.
+    #[inline]
+    #[must_use]
     #[stable(feature = "futures_api", since = "1.36.0")]
     #[rustc_const_unstable(feature = "const_waker", issue = "102012")]
-    #[must_use]
-    #[inline]
     pub const fn waker(&self) -> &'a Waker {
-        &self.waker
+        &self
+            .waker
+            .expect("no waker was set on this context, consider calling `local_waker` instead.")
```

Comment:
I don't think this is a good idea.

All existing code that uses `Future`s depend on this function returning `Waker`. So any `Context` without a `Waker` will cause this to panic.

Reply:
Well, the argument is that if a runtime decides not to support `Waker`, they will have to create a waker that panics on `wake()` instead of panicking in the call to `waker()`. Now, several people have expressed the same concern, so I do think it might be best to keep them mandatory for now. It would not be a breaking change to make them optional in the future, so we really don't need to commit to this right now.